cupy/_environment.py

```diff
@@ -476,10 +476,9 @@ def _detect_duplicate_installation():
         'cupy-rocm-4-2',
         'cupy-rocm-4-3',
         'cupy-rocm-5-0',
-    ]
-    cupy_installed = [
-        name for name in known
-        if list(importlib.metadata.distributions(name=name))]
+    }
+    installed_names = {d.name for d in importlib.metadata.distributions()}
```

Comment:
Fix for older Python
```suggestion
    installed_names = {d.metadata["Name"] for d in importlib.metadata.distributions()}
```